### PR TITLE
Test the `image.go` file

### PR DESCRIPTION
**Pull request by pPrecel**
**Description**

Changes proposed in this pull request:

- add tests to cover all the most important cases in the `image.go` file

**Related issue(s)**

- https://github.com/kyma-project/cli/issues/2001

### CODE REVIEW

File: internal/registry/image.go [hunk 1]
```diff
@@ -23,13 +24,28 @@ type ImportOptions struct {
 	RegistryPodPort      string
 }
 
+// for testing
+type utils struct {
+	daemonImage        func(name.Reference, ...daemon.Option) (v1.Image, error)
+	portforwardNewDial func(config *rest.Config, podName, podNamespace string) (httpstream.Connection, error)
+	remoteWrite        func(ref name.Reference, img v1.Image, options ...remote.Option) (rerr error)
```
Comment (anoipm) [inline internal/registry/image.go:31]: Do we need name of returned value (`rerr`) in this place?

File: internal/registry/image_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+						mock := automock.NewConnection(t)
+						mock.On("Close").Return(nil).Once()
+						return mock, nil
+					},
+				},
+			},
+			wantErr: errors.New("failed to push image to the in-cluster registry: registries must be valid RFC 3986 URI authorities: <    >"),
+		},
+		{
+			name: "import image",
```
Comment (anoipm) [inline internal/registry/image_test.go:141]: Does this test import image?

File: internal/registry/image.go (hunk 1 above)
Comment (pPrecel) [reply internal/registry/image.go:31]: nope, it's autogenerated. I will remove it

File: internal/registry/image_test.go (hunk 2 above)
Comment (pPrecel) [reply internal/registry/image_test.go:141]: at least tries but then fails with an error :P I will rename it

**Review by anoipm — APPROVED** [review]

**Pull request merged**